bundle/CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING 
+      "Choose the type of build, options are: Debug Release RelWithDebInfo." FORCE)
+endif()
+
+if(NOT BUILD_SHARED_LIBS)
+  set(BUILD_SHARED_LIBS ON CACHE BOOL "Build shared libraries." FORCE)
+endif()
+
+if(NOT CMAKE_INSTALL_PREFIX)
+  set(CMAKE_INSTALL_PREFIX "${CMAKE_SOURCE_DIR}/install" CACHE STRING
+      "Install path prefix, prepended onto install directories." FORCE)
+endif()
+
+project(dawn C CXX)
+enable_testing()
+cmake_minimum_required(VERSION 3.3)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/../cmake")
+
+include(DawnCloneRepository)
+
+set(MB_GIT_URL "https:///github.com/cosunae/msbuild.git"
+    CACHE PATH "URL of the dawn git repository to clone")
+set(MB_GIT_BRANCH "master" CACHE STRING "Branch of the dawn git repository to clone")
+
+dawn_clone_repository(NAME msbuild URL ${MB_GIT_URL} BRANCH ${MB_GIT_BRANCH} SOURCE_DIR DAWN_MSBUILD_SOURCE_DIR )
+
+#
+list(APPEND CMAKE_MODULE_PATH "${DAWN_MSBUILD_SOURCE_DIR}/cmake")
+include(msbuildCMakeInit)
+
+include(msbuildAddTargetCleanAll)
+include(msbuildCheckInSourceBuild)
+include(msbuildGetArchitectureInfo)
+include(msbuildGetCompilerInfo)
+include(msbuildGetPlatformInfo)
+include(msbuildMakeStringPair)
+include(msbuildReportResult)
+
+# Include all options (including all Dawn and GTClang options)
+#include(GTClangAllOptions)
+
+msbuild_get_compiler_info()
+msbuild_get_platform_info()
+msbuild_get_architecture_info()
+
+# Prevent in source builds
+msbuild_check_in_source_build()
+
+# Add custom targets
+msbuild_add_target_clean_all(
+  "${CMAKE_BINARY_DIR}/prefix"
+  "${CMAKE_BINARY_DIR}/thirdparty"  
+  "${CMAKE_BINARY_DIR}/dawn-cmake"  
+  "${CMAKE_BINARY_DIR}/dawn"
+  "${CMAKE_BINARY_DIR}/dawn-prefix"
+  "${CMAKE_BINARY_DIR}/gtclang"
+  "${CMAKE_BINARY_DIR}/gtclang-prefix"
+  "${CMAKE_BINARY_DIR}/Makefile"
+  "${CMAKE_BINARY_DIR}/msbuild-cmake"
+  "${CMAKE_BINARY_DIR}/protobuf"
+  "${CMAKE_BINARY_DIR}/protobuf-prefix"
+)
+
+# Output summary of the configuration
+macro(make_config_string FIRST SECOND)
+  msbuild_make_string_pair(${FIRST} ": ${SECOND}" 20 out)
+  list(APPEND config_info ${out})
+endmacro()
+make_config_string("Platform" ${MSBUILD_PLATFORM_STRING})
+make_config_string("Architecture" ${MSBUILD_ARCHITECTURE_STRING})
+make_config_string("Compiler" ${MSBUILD_COMPILER_STRING})
+make_config_string("Build type" ${CMAKE_BUILD_TYPE})
+msbuild_report_result("Configuration summary" ${config_info})
+
+include(DawnAddDependency)
+
+dawn_add_dependency(NAME "protobuf")
+
+
+set(dawn_cmake_args -DProtobuf_DIR=${Protobuf_DIR})
+
+msbuild_find_package(
+  PACKAGE dawn
+  FORWARD_VARS 
+    BINARY_DIR dawn_binary_dir
+  DEPENDS "protobuf"
+  ADDITIONAL
+    SOURCE_DIR "${CMAKE_SOURCE_DIR}/../"
+    MSBUILD_ROOT "${DAWN_MSBUILD_SOURCE_DIR}"
+    CMAKE_ARGS 
+        ${dawn_cmake_args}
+)
+
+if(NOT(dawn_binary_dir))

```

Comment:
sorry this was not clear (and also not yet documented).  dawn_binary_dir is not setby the user, but by the find_package. It is the varable that indicates where is the build directory, so that we can call the ctest from there. The idea is that the FORWARD_VARS will tell find_package that , in case we need to build the package, they need to be passed to External_pkg.cmake and set properly. There are checks at several levels that all FORWARD_VARS are set. The idea was also to not used a convention, like expect that the External_pkg.cmake will put that in certain fixed global variable, since we are then connecting too much functions via global variables. Instead we pass output variables that need to be set.

Reply:
what does `if(NOT(var))` ? does it check that the variable is defined and the string is not empty? 
That is what I intended, since is better than checking that is defined (but could be empty string)... but I am not sure